### PR TITLE
Add connection info to title

### LINKED ISSUE #1193: Add displayed current connection
- SQL Operations Studio Version:
Version: 0.28.3-insider

Steps to Reproduce:
image

like that
image

### PR DESCRIPTION
**Pull request by anthonydresser**
I'm not sure if this is the right look, but it implements what ssms has. fixes #1193
image
image
connectionuser

_______________________________
added title shortening
image

### CODE REVIEW

**Comment by kburtram** [comment]
@anthonydresser it looks like SSMS has some logic to shorten the name with "..." to keep the tabs from getting too wide.  I'd recommend we do something similar otherwise I think the tabs will be too wide.

For example, here's what SSMS does.  Thoughts?

screen

**Comment by kburtram** [comment]
@anthonydresser I couldn't find the code that actual does the title shorten since it's buried in Visual Studio shell code.  But I could see the code that called the formatting code and the logic is something like the following.

1. A document caption is multiple parts such as (a) solution name (b) document name (c) document additional info.
2. SSMS passes connection info as the additional info parameter
3. "Built-in" caption components like solution or document name aren't shortened
4. Additional info will be capped at a certain character count.  The first few characters, then `...`, then the last several characters will be displayed if the additional info is larger max allowed char count.

I think we can apply something similar here probably.  Particularly if VS Code doesn't want to address the issue upstream.

**Comment by anthonydresser** [comment]
@kburtram Well I also noticed that it doesn't worry about making the authentication info shortened, it seems to ignore that part during shortening.

**Comment by kburtram** [comment]
@anthonydresser perhaps if you try a longer username?  For example, with `thisisareallylongusernamethatmaybetruncatedifitstoolong` I observe the below.  Please let me know if I misunderstand your scenario.

screen

**Comment by anthonydresser** [comment]
I added the title shortening, I basically just remove the middle of the connection info to aim at a max size of 13 characters.

**Comment by anthonydresser** [comment]
@kburtram @yualan I'm thinking it might be worth to put this behind a settings (showing connection info in the title), thoughts?

File: src/vs/workbench/electron-browser/main.contribution.ts [hunk 1]
```diff
@@ -288,6 +288,12 @@ configurationRegistry.registerConfiguration({
 			'type': 'boolean',
 			'description': nls.localize('enableNaturalLanguageSettingsSearch', "Controls whether to enable the natural language search mode for settings."),
 			'default': true
+		},
+		// {{SQL CARBON EDIT}}
+		'workbench.editor.showConnectionInfoInTitle': {
```
Comment (kburtram) [inline src/vs/workbench/electron-browser/main.contribution.ts]: it is possible to add this setting from under the `sql` directory somewhere?  I'd refer to keep the `vs` changes minimized as much as possible, particularly for SQL-related changes like this that we wouldn't consider pushing upstream at some point.

Comment (anthonydresser) [reply src/vs/workbench/electron-browser/main.contribution.ts]: I was thinking from a user point of view, i'm not going to know the different between sql settings and non-sql settings, so putting a tab title settings near all the other tab title settings made the most sense from a user perspective.

**Pull request merged**